music_assistant/server/providers/apple_music/__init__.py

```diff
@@ -237,7 +257,16 @@ async def get_album_tracks(self, prov_album_id) -> list[Track]:
         """Get all album tracks for given album id."""
         endpoint = f"catalog/{self._storefront}/albums/{prov_album_id}/tracks"
         response = await self._get_data(endpoint, include="artists")
-        return [self._parse_track(track) for track in response["data"] if track["id"]]
+        # Including albums results in a 504 error, so we need to fetch the album separately
```

Comment:
How does this solve that 504 error ?

Reply:
`include="artists,albums"` messes with the API and throws a 504. This should be supported according to the docs and definitely works for other endpoints, but somehow crashes on the `/albums/xxx/tracks` endpoint. Using either `include="artists"` or `include="albums"` seems to work fine. Since we need both, I opted to retrieve the Album object once and set it on each individual Track

Reply:
Yeah, my point was that there was no inclusion of both in the call so strictly taken this was not broken ?

Reply:
But both are required because tracks were missing an album, causing some weird behavior. To fix that, I needed to include both which threw a consistent 504, so I worked around that.